### PR TITLE
Filters for task added

### PR DESCRIPTION
**Pull request by Huzii-droid19**
Home screen features, such as filter task on the date selected and search query.
Code structure is also improved

### CODE REVIEW

File: src/components/Header/index.tsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+};
+const Header = ({
+  currentDate,
+  calendarHandler,
+  onSearch,
+  search,
+}: HeaderProps) => {
+  const [isSearchingVisible, setIsSearchingVisible] =
+    React.useState<boolean>(false);
+  const searchHandler = () => {
```
Comment (shoaib2529) [inline src/components/Header/index.tsx:70]: searchHandler isn't appropriate name for this function.
Also this function doesn't seems required as it has just one line in it.

File: src/components/Header/styles.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  justify-content: space-between;
+  align-self: flex-start;
+  padding-horizontal: 25px;
+  height: 10%;
+  width: 100%;
+  height: 10%;
+  elevation: 5;
+  background-color: ${({theme}: {theme: string}) => theme};
+`;
+export const MonthWrapper = styled(TouchableOpacity)`
```
Comment (shoaib2529) [inline src/components/Header/styles.ts:20]: This method is just for custom & third-party components, you shouldn't use it for react-native components. Instead use styled.TouchableOpacity

**Review by shoaib2529 — CHANGES_REQUESTED** [review]

File: src/components/Header/styles.ts (hunk 2 above)
Comment (Huzii-droid19) [reply src/components/Header/styles.ts:20]: sure, i'll resolve it

File: src/components/Header/index.tsx [hunk 3]
```diff
@@ -14,26 +14,27 @@ type SearchBarProps = {
...
-  searchHandler: () => void;
+  toggleSearch: (prevState: any) => void;
 };
 
 type MonthViewProps = {
   currentDate: string;
   calendarHandler: () => void;
   iconColor: string;
-  searchHandler: () => void;
+  toggleSearch: (prevState: any) => void;
```
Comment (shoaib2529) [inline src/components/Header/index.tsx:24]: `any` type shouldn't be used when you know the type.

**Review by shoaib2529 — CHANGES_REQUESTED** [review]

File: src/components/Header/index.tsx (hunk 3 above)
Comment (Huzii-droid19) [reply src/components/Header/index.tsx:24]: I'm specifying type to 'boolean', but it says cant assign boolean to boolean

Comment (Huzii-droid19) [reply src/components/Header/index.tsx]: any type changed to boolean

**Pull request merged**